source/dataman/CLS/CLSDbUpgrade1Pt1.cpp

```diff
@@ -0,0 +1,118 @@
+#include "CLSDbUpgrade1Pt1.h"
+
+#include <QStringBuilder>
+
+#include "util/AMErrorMonitor.h"
+
+CLSDbUpgrade1Pt1::CLSDbUpgrade1Pt1(const QString &facilityName, const QString &databaseNameToUpgrade, QObject *parent)
+	: AMDbUpgrade(databaseNameToUpgrade, parent)
+{
+	targetFacilityName_ = facilityName;
+}
+
+CLSDbUpgrade1Pt1::~CLSDbUpgrade1Pt1()
+{
+
+}
+
+QStringList CLSDbUpgrade1Pt1::upgradeFromTags() const
+{
+	return QStringList();
+}
+
+bool CLSDbUpgrade1Pt1::upgradeNecessary() const
+{
+	// run update when:
+	//
+	//   - the AMRun table has the "facilityId" column
+	//   - The AMFacility table has more than one record, or
+	//   - the name of the only facility doesn't match with the given beamline name, or
```

Comment:
I believe we were thinking of making BioXAS just one facility a while ago.  Would it make sense?

To answer your question though, I don't believe it would matter for this database upgrade the way it is organized.